assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,153 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useSelect } from '@wordpress/data';
+import { ruleCard } from './styles';
+
+
+const Rule = ({
+  busy,
+  field,
+  onDelete,
+  onUpdate,
+  reorderRule,
+  ruleIndex,
+  setOriginIndex,
+  setTargetIndex,
+  taxonomy,
+  term_id,
+  value,
+}) => {
+  const {
+    fields,
+    sections,
+    taxonomies,
+    themes,
+  } = wpLocalizedData;
+
+  // const { loadingTerms, taxTerms } = useSelect((select) => ({
+  //   loadingTerms: select('core/data').isResolving('core', 'getEntityRecords', ['taxonomy', 'category']),
+  //   taxTerms: select('core').getEntityRecords('taxonomy', 'category') || [],
+  // }));
+  // if(!loadingTerms) {
+  //   console.log(taxTerms)
+  // }
+
+  return (
+    <div
+      className="rule-wrapper"
+      draggable
+      style={ruleCard}
+      onDragEnd={(e) => {
+        const targetEl = document.elementFromPoint(e.clientX, e.clientY);
+        // Only reorder if the target element is inside rule flex container.
+        if (targetEl.closest('.rule-wrapper')) {
+          reorderRule();
+        }
+      }}
+      onDragStart={() => setOriginIndex(ruleIndex)}
+      onDragOver={(e) => {
+        e.preventDefault();
+        setTargetIndex(ruleIndex)
+      }}
+    >
+      <SelectControl
+        disabled={busy}
+        label={__('Taxonomy', 'apple-news')}
+        onChange={(next) => onUpdate(ruleIndex, 'taxonomy', next)}
```

Comment:
After my suggestions above are applied this and the other `onChange` items can be updated to not include the index, just the key and value, like:
```suggestion
        onChange={(next) => onUpdate('taxonomy', next)}
```